satellite/satellitedb/overlaycache.go

```diff
@@ -902,6 +902,55 @@ func (cache *overlaycache) UpdateUptime(ctx context.Context, nodeID storj.NodeID
 	return getNodeStats(dbNode), Error.Wrap(tx.Commit())
 }
 
+// AllPieceCounts returns a map of node IDs to piece counts from the db.
+func (cache *overlaycache) AllPieceCounts(ctx context.Context) (_ map[storj.NodeID]int, err error) {
+	defer mon.Task()(&ctx, err)
+
+	// NB: `All_Node_Id_Node_PieceCount_By_PieceCount_Not_Number` selects node
+	// ID and piece count from the nodes table where piece count is not zero.
+	rows, err := cache.db.All_Node_Id_Node_PieceCount_By_PieceCount_Not_Number(ctx)
+	if err != nil {
+		return make(map[storj.NodeID]int), Error.Wrap(err)
```

Comment:
I would say that the most common convention in Go is when an error is returned the other return parameters should be ignored; when it isn't respected the documentation of the function should have a comment for it.

I personally prefer that this function return nil when an error and it's the caller which instantiates a map if it requires it. However, I'm happy about leaving as it's and commenting it in the function documentation.